optaplanner-core/src/main/java/org/optaplanner/core/impl/solver/manager/DefaultSolverManager.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.core.impl.solver.manager;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadFactory;
+import java.util.function.Consumer;
+
+import org.optaplanner.core.api.score.Score;
+import org.optaplanner.core.api.solver.SolverFactory;
+import org.optaplanner.core.api.solver.manager.SolverManager;
+import org.optaplanner.core.api.solver.manager.SolverStatus;
+import org.optaplanner.core.impl.solver.thread.ThreadUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DefaultSolverManager<Solution_> implements SolverManager<Solution_> {
+
+    private static final Logger logger = LoggerFactory.getLogger(DefaultSolverManager.class);
+
+    private ExecutorService solverExecutorService;
+    private ExecutorService eventHandlerExecutorService;
+    private SolverFactory<Solution_> solverFactory;
+    private ConcurrentMap<Object, SolverTask<Solution_>> problemIdToSolverTaskMap;
+
+    public DefaultSolverManager(String solverConfigResource) {
+        this(solverConfigResource, null, null);
+    }
+
+    public DefaultSolverManager(String solverConfigResource, ClassLoader classLoader) {
+        this(solverConfigResource, classLoader, null);
+    }
+
+    public DefaultSolverManager(String solverConfigResource, ThreadFactory threadFactory) {
```

Comment:
And these public constructors should require non-null arguments. Tip: use `Objects.requireNonNull()`.